ansys/dpf/core/plotter.py

```diff
@@ -693,13 +693,22 @@ def plot_contour(
                 unit = field.unit
                 break
 
+        # If ElementalNodal, first extend results to mid-nodes
+        if location == locations.elemental_nodal:
+            fields_container = dpf.core.operators.averaging.extend_to_mid_nodes_fc(
+                fields_container=fields_container
+            ).eval()
+
+        location_data_len = mesh.location_data_len(location)  # 3751 nodes  # 3000 elements  # 24000 elemental nodal  # field.data 72000
         if location == locations.nodal:
             mesh_location = mesh.nodes
         elif location == locations.elemental:
             mesh_location = mesh.elements
+        elif location == locations.elemental_nodal:
+            mesh_location = mesh.elements
         else:
             raise ValueError(
-                "Only elemental or nodal location are supported for plotting."
+                "Only elemental, elemental nodal or nodal location are supported for plotting."
```

Comment:
```suggestion
                "Only elemental, elemental nodal, or nodal location are supported for plotting."
```
We use the Oxford comma, which is to always place a comma before the "and" or the "or" in a list with three or or more items.